### PR TITLE
Fixes Hookshots and adds Sanity to exiting certain machines through unconventional means

### LINKED ISSUE #33183: Duplicate bodies (?) by hookshotting them out of cryo
(WEB REPORT BY: poales REMOTE: 206.221.180.138:7777)
# Revision
fd808d68d314904b39ca913c8f692e1c5eba8f72

# Description
if you hookshot someone, put them in a cryo tube, then pull them out with the hookshot their body gets visually duplicated, while their spirit gets (apparently) sent to gay baby jail

# Steps to Reproduce
Hookshot
put in toob
walk away
put the second body in the second toob
walk away
have a third body

# What you Expected
I thought going into cryo would break the hookshot

# What Actually Happened
admins had to teleport his ghost back into the world of the living

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #33183

https://github.com/vgstation-coders/vgstation13/assets/7573912/130ad3c3-4c8d-4e8d-9526-6a08e3ccb35c

Hookshots, as well as cult and AI jaunts now use transform matrices to rotate themselves instead of the Turn() icon proc, bringing them in line with the rest of projectiles for the past 5 years.

:cl:
* bugfix: Fixed hookshots and their chains not being visible in many cases.
* bugfix: Fixed hookshots having a much shorter range when firing in diagonals.
* bugfix: Fixed hookshots failing to properly reel you in if the hook hits something at the longest range possible.
* bugfix: Fixed a sanity issue that occurred when mobs exit a Cryo Tube, Body Scanner, DNA Scanner, Mecha, or Cyborg Recharging Station through teleportation or getting pulled by a hookshot chain, and that would either allow other players to teleport them back to it, or just prevent the object from being used by other players.

### CODE REVIEW

File: code/game/machinery/adv_med.dm [hunk 1]
```diff
@@ -176,6 +176,11 @@
 		set_light(light_range_on, light_power_on)
 	return
 
+/obj/machinery/bodyscanner/Exited(var/atom/movable/O) // Used for teleportation from within the scanner.
```
Comment (ShiftyRail) [inline code/game/machinery/adv_med.dm:179]: else return ..() ?

File: code/game/machinery/cryo.dm [hunk 2]
```diff
@@ -499,6 +499,14 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
 	//expel_gas.temperature = T20C // Lets expel hot gas and see if that helps people not die as they are removed
 	//loc.assume_air(expel_gas)
 
+/obj/machinery/atmospherics/unary/cryo_cell/Exited(var/atom/movable/O) // Used for teleportation from within the tube.
```
Comment (ShiftyRail) [inline code/game/machinery/cryo.dm:502]: else return ..() ?

File: code/game/machinery/rechargestation.dm [hunk 3]
```diff
@@ -247,13 +247,18 @@
 	capacitor_stored = min(capacitor_stored + (20 * transfer_rate_coeff), capacitor_max)
 	return 1
 
+/obj/machinery/recharge_station/Exited(var/atom/movable/O) // Used for teleportation from within the recharge station.
```
Comment (ShiftyRail) [inline code/game/machinery/rechargestation.dm:250]: else return ..() ?

File: code/game/mecha/mecha.dm [hunk 4]
```diff
@@ -1373,6 +1373,17 @@
 			continue
 		O.forceMove(loc) //Somehow got inside, drop it.
 
+/obj/mecha/Exited(var/atom/movable/O) // Used for teleportation from within the mecha.
```
Comment (ShiftyRail) [inline code/game/mecha/mecha.dm:1376]: else return ..() ?

**Review by ShiftyRail — COMMENTED** [review]
I actually do not know if the parent of these does anything fancy, but it seems like something that should be done.

**Comment by DeityLink** [comment]
future proof'd

**Pull request merged**